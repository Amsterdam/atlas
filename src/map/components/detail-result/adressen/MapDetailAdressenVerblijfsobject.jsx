import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultStatusItem from '../MapDetailResultStatusItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';
import MapDetailAdressenVerblijfsobjectGebruiksdoelenItem from './MapDetailAdressenVerblijfsobjectGebruiksdoelenItem';
import Notification from '../../../../shared/components/notification/Notification';

const statusToCssModifier = {
  18: 'alert',
  19: 'alert',
  20: '',
  21: '',
  22: 'alert',
  23: 'alert'
};

const MapDetailAdressenVerblijfsobject = ({
  panoUrl, verblijfsobject, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={verblijfsobject.label}
    title="Adres"
  >
    <ul className="map-detail-result__list">
      { statusToCssModifier[verblijfsobject.status.code] && <li>
        <Notification level="alert">Status: {verblijfsobject.status.description}</Notification>
      </li> }
      { verblijfsobject.isNevenadres && <li>
        <Notification level="info">Dit is een nevenadres</Notification>
      </li> }
      { verblijfsobject.gebruiksdoelen && <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={verblijfsobject.gebruiksdoelen}
      /> }
      <MapDetailResultItem
        label="Oppervlakte"
        value={verblijfsobject.size ? `${verblijfsobject.size} mÂ²` : 'onbekend'}
      />
      <MapDetailResultItem
        label="Type woonobject"
        value={verblijfsobject.type}
      />
      <MapDetailResultItem
        label="Eigendomsverhouding"
        value={verblijfsobject.eigendomsverhouding}
      />
      <MapDetailResultStatusItem
        label="Indicatie geconstateerd"
        value={verblijfsobject.indicatieGeconstateerd ? 'Ja' : 'Nee'}
        status={verblijfsobject.indicatieGeconstateerd ? 'alert' : ''}
      />
      <MapDetailResultStatusItem
        label="Indicatie hoofdadres"
        value={verblijfsobject.isNevenadres ? 'Nee' : 'Ja'}
        status={verblijfsobject.isNevenadres ? 'info' : ''}
      />
      <MapDetailResultStatusItem
        label="Status"
        value={verblijfsobject.status.description}
        status={statusToCssModifier[verblijfsobject.status.code]}
      />
      <MapDetailResultStatusItem
        label="Aanduiding in onderzoek"
        value={verblijfsobject.aanduidingInOnderzoek ? 'Ja' : 'Nee'}
        status={verblijfsobject.aanduidingInOnderzoek ? 'alert' : ''}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailAdressenVerblijfsobject.propTypes = {
  verblijfsobject: PropTypes.shape({
    aanduidingInOnderzoek: PropTypes.boolean,
    eigendomsverhouding: PropTypes.string,
    gebruiksdoelen: PropTypes.array,
    indicatieGeconstateerd: PropTypes.boolean,
    isNevenadres: PropTypes.boolean,
    label: PropTypes.string,
    size: PropTypes.number,
    type: PropTypes.string,
    status: PropTypes.shape({
      description: PropTypes.string,
      code: PropTypes.string
    })
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailAdressenVerblijfsobject;
