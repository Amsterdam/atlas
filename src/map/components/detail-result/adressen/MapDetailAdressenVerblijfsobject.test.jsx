import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAdressenVerblijfsobject from './MapDetailAdressenVerblijfsobject';

describe('MapDetailAdressenVerblijfsobject', () => {
  it('should render everything', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      size: 15,
      type: 'Verblijfsobject type'
    };
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown with zero value for size', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      size: 0,
      type: 'Verblijfsobject type'
    };
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown without size', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      type: 'Verblijfsobject type'
    };
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
