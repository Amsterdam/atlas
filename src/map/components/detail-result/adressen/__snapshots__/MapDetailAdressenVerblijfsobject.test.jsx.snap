// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAdressenVerblijfsobject rendering should not render size 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        Adres
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        Verblijfsobject label
      </h2>
    </div>
  </header>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value=""
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
  </ul>
</section>
`;

exports[`MapDetailAdressenVerblijfsobject rendering should render everything 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--pano
      "
  >
    <img
      alt="Panoramabeeld"
      className="map-detail-result__header-pano"
      src="panoUrl"
    />
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        Adres
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        Verblijfsobject label
      </h2>
    </div>
  </header>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="15 m²"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
  </ul>
</section>
`;

exports[`MapDetailAdressenVerblijfsobject rendering should render with zero value for size 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        Adres
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        Verblijfsobject label
      </h2>
    </div>
  </header>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="0 m²"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
  </ul>
</section>
`;

exports[`MapDetailAdressenVerblijfsobject rendering should render without pano url 1`] = `
<section
  className="map-detail-result"
>
  <header
    className="
        map-detail-result__header
        map-detail-result__header--no-pano
      "
  >
    <div
      className="map-detail-result__header-container"
    >
      <h1
        className="map-detail-result__header-title"
      >
        Adres
      </h1>
      <h2
        className="map-detail-result__header-subtitle"
      >
        Verblijfsobject label
      </h2>
    </div>
  </header>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="15 m²"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
  </ul>
</section>
`;
